Code/PgSQL/rdkit/mol_op.c

```diff
@@ -376,7 +378,7 @@ fmcs_mol2s_transition(PG_FUNCTION_ARGS) {
     }
     if(PG_ARGISNULL(0) && !PG_ARGISNULL(1)) { // first call
 ///elog(WARNING, "fmcs_mol2s_transition() called first time");
-        CROMol mol = PG_GETARG_DATUM(1);
+        CROMol mol = (CROMol)PG_GETARG_DATUM(1);
```

Comment:
If you are worried about limits, you can use rdcast which will fail in debug mode for numeric conversions that don't work (i.e. -1 -> unsigned ).  It does a static cast in non debug mode.